dallinger/docker/tools.py

```diff
@@ -200,11 +198,11 @@ def run_compose(self, compose_commands: str):
         )
 
 
-class DockerStartupError(RuntimeError):
+class DockerStartupError(click.Abort):
```

Comment:
This is probably a bit nit-picky, but I think we should probably only raise click errors from click commands. It seems to me like we should throw a non-click error in the utils/tools, catch and log it in the click command and then raise the Abort.

Reply:
I agree that a more pure approach would be desirable, but I believe in this case inheriting from `click.Abort` allows for less code duplication.

The `BuildError` exception in particular might be raised during a `dallinger docker build` command (that directly triggered the build) or during a `dallinger docker debug` run (that might start building a container in case the image has not been built yet).

For these reasons I'd rather keep code as is instead of amending it as suggested.